scene/gui/scroll_container.cpp

```diff
@@ -244,6 +248,16 @@ void ScrollContainer::gui_input(const Ref<InputEvent> &p_gui_input) {
 	}
 }
 
+void ScrollContainer::set_max_size(Size2 p_max_size) {
+	max_size = p_max_size.max(Vector2());
```

Comment:
```suggestion
	if (max_size == p_max_size) {
		return;
	}
	ERR_FAIL_COND(p_max_size.height < 0 || p_max_size.width < 0);
```

`queue_sort` and `update_minimum_size` are expensive, there's no reason to run it if nothing changes. Also, invalid value probably should show an error instead of silently using `max`.